war/src/main/js/add-item.js

```diff
@@ -0,0 +1,293 @@
+    
+    //////////////////////////////
+    // helpful reference DOM
+    
+    var jRoot = $('head').attr('data-rooturl');
+    var defaultMinToShow = 2;
+    var $root = $jq('#main-panel');
+    var $form = $root.find('form[name="createItem"]').addClass('jenkins-config new-view');
+    var $newView = $jq('<div class="new-view" />')
+      .attr('name','createItem')
+      .attr('action','craetItem')
+      .prependTo($form);
+    var $tabs = $('<div class="jenkins-config-widgets" />').appendTo($newView);
+    var $categories = $('<div class="categories" />').appendTo($newView);
+    var sectionsToShow = [];    
+
+    
+    ////////////////////////////////
+    // scroll action......
+    
+    function watchScroll(){
+      var $window = $(window);
+      var $jenkTools = $('#breadcrumbBar');
+      var winScoll = $window.scrollTop();
+      var jenkToolOffset = $jenkTools.height() + $jenkTools.offset().top + 15;
+
+      $tabs.find('.active').removeClass('active');
+      $.each(data.categories,function(i,cat){
+        var domId = '#j-add-item-type-'+cat.id;
+        var $cat = $(domId);
+        var catHeight = ($cat.length > 0)?
+            $cat.offset().top + $cat.outerHeight() - (jenkToolOffset + 100):
+              0;
+
+        if(winScoll < catHeight){
+          var $thisTab = $tabs.find(['[href="',cleanHref(domId),'"]'].join(''));
+          resetActiveTab($thisTab);
+          return false;
+        }
+      });
+      
+      if(winScoll > $('#page-head').height() - 5 ){  
+        $tabs.width($tabs.width()).css({
+          'position':'fixed',
+          'top':($jenkTools.height() - 5 )+'px'});
+        $categories.css({'margin-top':$tabs.outerHeight()+'px'});   
+      }
+      else{
+        $tabs.add($categories).removeAttr('style');
+      }
+    }
+
+    //////////////////////////
+    // helper functions...
+    
+    function addCopyOption(data){
+      var $copy = $('#copy').closest('tr');
+      if($copy.length === 0) {return data;} // exit if copy should not be added to page. Jelly page holds that logic.
+      var copyTitle = $copy.find('label').text();
+      var copyDom = $copy.next().find('.setting-main').html();
+      var copy = {
+          name:'Copy',
+          id:'copy',
+          minToShow:0,
+          weight:-999999999999, ///some really big number so it can be last...
+          items:[
+            {
+              class:"copy",
+              description:copyDom,
+              displayName:copyTitle,
+            }
+          ]
+      };
+      var newData = [];
+      
+      $.each(data,function(i,elem){
+        if(elem.id !== "category-id-copy")
+          { newData.push(elem); }
+      });
+      
+      newData.push(copy);
+
+      
+      return newData;
+    }
+    
+    function sortItemsByOrder(itemTypes) {
+      function sortByOrder(a, b) {
+        var aOrder = a.weight;
+        var bOrder = b.weight;
+        return ( (aOrder < bOrder) ? -1 : ( (aOrder > bOrder) ? 1 : 0));
```

Comment:
This will balls up with nulls, I suggest changing the second comparison to === rather than &gt; and reversing the 1:0